src/gepa/adapters/generic_rag_adapter/GEPA_RAG.md

```diff
@@ -0,0 +1,621 @@
+- **Comprehensive Evaluation**: Both retrieval quality (precision, recall, MRR) and generation quality (F1, BLEU, faithfulness) metrics
+- **Production Ready**: Battle-tested with proper error handling, logging, and performance monitoring
+- **5 Vector Stores Supported**: Complete examples for ChromaDB, Weaviate, Qdrant, Milvus, and LanceDB
+
+## 🚀 Quick Start
+
+### Installation
+
+```bash
+# Install core GEPA package
+pip install gepa
+
+# Install RAG adapter dependencies
+# Navigate to the examples/rag_adapter directory
+cd src/gepa/examples/rag_adapter
+
+# Option A: Install all vector store dependencies (recommended for exploration)
+pip install -r requirements-rag.txt
+
+# Option B: Install specific vector store dependencies
+pip install litellm chromadb                    # For ChromaDB
+pip install litellm weaviate-client             # For Weaviate  
+pip install litellm lancedb pyarrow             # For LanceDB
+pip install litellm pymilvus                    # For Milvus
+pip install litellm qdrant-client               # For Qdrant
+
+# Setup local Ollama models for examples
+ollama pull qwen3:8b          # Default for ChromaDB/Weaviate/Qdrant
+ollama pull llama3.1:8b       # Default for LanceDB/Milvus
+ollama pull nomic-embed-text:latest  # Embedding model
+```
+
+**Note:** For specific version requirements, see the `requirements-rag.txt` file in the `examples/rag_adapter/` directory.
+
+### 5-Minute Example
+
+```python
+import gepa
+from gepa.adapters.generic_rag_adapter import GenericRAGAdapter, ChromaVectorStore
+
+# 1. Setup your vector store
+vector_store = ChromaVectorStore.create_local("./knowledge_base", "documents")
+
+# 2. Create the RAG adapter
+adapter = GenericRAGAdapter(
+    vector_store=vector_store,
+    llm_model="ollama/llama3.2:1b",  # Memory-friendly local model
+    # llm_model="gpt-4",  # For cloud-based models (requires API key)
+    rag_config={
+        "retrieval_strategy": "similarity",
+        "top_k": 3  # Reduced for faster local testing
+    }
+)
+
+# 3. Define your training data
+train_data = [
+    {
+        "query": "What is machine learning?",
+        "ground_truth_answer": "Machine learning is...",
+        "relevant_doc_ids": ["doc_001"],
+        "metadata": {"category": "AI"}
+    }
+    # ... more examples
+]
+
+# 4. Define initial prompts to optimize
+initial_prompts = {
+    "answer_generation": "Answer the question based on the provided context.",
+    "context_synthesis": "Synthesize the following documents into a coherent context."
+}
+
+# 5. Run GEPA optimization (local-friendly settings)
+result = gepa.optimize(
+    seed_candidate=initial_prompts,
+    trainset=train_data,
+    valset=validation_data,
+    adapter=adapter,
+    max_metric_calls=10,  # Small number for local testing
+    reflection_llm_model="ollama/llama3.1:8b"  # Memory-friendly reflection model
```

Comment:
Could you update the reflection_llm to one of the more recent models that can run locally? For example, qwen3-8B is better than Llama-3-8B for reflection.